### DISCUSSION TITLE
captchaExpiryTimeLimit in secs?

### DISCUSSION THREAD

**Discussion by kringo** [librecaptcha/lc-core#100] [Q&A] [answered]
What's the value for captchaExpiryTimeLimit should be, in secs?

If so, i've set it to 60 but after submitting a wrong answer, typing the correct answer again I get result "Expired"

Wondering how to get this right?

**Comment by hrj** [comment, marked as answer]
Please feel free to ask more questions. Copying my answer from my previous reply: 

Are you re-attempting with the same token? That will always fail, otherwise bots will get multiple chances to try different guesses for the same token. After a failed attempt you need to ask the API for a fresh CAPTCHA challenge.

The way this works is:
The server maintains a set of CAPTCHA challenges. When a new captcha is requested from the API, it creates a UUID for the challenge that can only be attempted once. After an attempt is made for that UUID, that UUID is deleted, and internally an attempt count is incremented for the CAPTCHA.

When a CAPTCHA has been attempted many times (according to the internal counter), that CAPTCHA is retired, and a new CAPTCHA is generated to replace it.

**Comment by kringo** [comment]
Thanks @hrj

**Comment by hrj** [comment]
Sorry, I missed this part:

> What's the value for captchaExpiryTimeLimit should be, in secs?

I think it is in minutes.
